lstchain/reco/r0_to_dl1.py

```diff
@@ -188,9 +187,16 @@ def r0_to_dl1(
 
     """
     if output_filename is None:
-        output_filename = (
-            'dl1_' + os.path.basename(input_filename).rsplit('.', 1)[0] + '.h5'
-        )
+        if (input_filename.startswith('LST')):
+            output_filename = (
```

Comment:
This looks like you are expecting a certain filename pattern here. Not just that the file starts with `LST`.

What is the format?

Reply:
Yes, names are e.g.:
`LST-1.4.Run01975.0005.fits.fz`
so:
'LST-' + `tel_id` + '.' + `stream_number` + '.Run' + `run_number` + '.' + `subrun_number` + '.fits.fz'

Reply:
So I think we should use a regex for this and check if this matches, I'll prepare a PR.

Reply:
And you want to remove the stream here? So dl1_LST-{tel_id}.{run_number}.{stream_number}'?

Reply:
No, that is the misleading thing about streams, see #301 . 
The reader, if all 4 streams are in the same folder, reads the 4 of them and aligns the event numbers from the four files. So when you process the data in the standard way, event though the input string may be `LST-1.1.Run01975.0005.fits.fz`, the `dl1` output has all 4 streams, that was the reason to get rid of the `stream_number`

Reply:
I agree that this may be problematic if you process only one of the streams, because you would lose the stream number in the output, but this is related to how the reader processes the data and in any case, you process all the 4 streams in most of the cases

Reply:
THis is what I wrote here, or not? The intention of the code is to add the `dl1_` prefix, replace the extension with `.h5` and to remove the stream  number, correct?

Reply:
yes, that is right, but this is what is done in the code right now, right?
sorry, I read the original message that contained some `stream_number` in the output name